### ISSUE TITLE
wrong step into/over icon color before the focus is on the repl window

### ISSUE THREAD

**Issue by st0nie** [rcarriga/nvim-dap-ui#258]
![Screenshot_20230404_001430](https://user-images.githubusercontent.com/42872734/229567959-02f5afd0-38c6-4143-adb8-dd549fb82272.png)
dap-ui is really good to use, but when I used it recently, I found that the icons of some operations turned gray, making them difficult to read, and I don't know how to change the color of these icons.

**Comment by EpsilonKu** [comment]
It's gray because u can't use it currently. When process stop on breakpoint, it will turn to green

**Comment by st0nie** [comment]
> It's gray because u can't use it currently. When process stop on breakpoint, it will turn to green
![Screenshot_20230404_013726](https://user-images.githubusercontent.com/42872734/229585426-1f1808ea-c9d0-451a-b7b7-a372a4d2ce01.png)
It doesn't seem that way, I've stopped at the breakpoint now, but the icon is still grey, while I can click on the icon and it also has the correct effect

**Comment by st0nie** [comment]
> It's gray because u can't use it currently. When process stop on breakpoint, it will turn to green
oops.. so sorry this is my problem, when i move the focus to dap-repl's window, it does turn blue, but if i keep the cursor on another window, no matter how i click on the icon, it won't turn blue

**Comment by st0nie** [comment]
oops.. so sorry this is my problem, when i move the focus to dap-repl's window, it does turn blue, but if i keep the cursor on another window, no matter how i click on the icon, it won't turn blue

**Comment by st0nie** [comment]
> It's gray because u can't use it currently. When process stop on breakpoint, it will turn to green

oops.. so sorry this is my problem, when i move the focus to dap-repl's window, it does turn blue, but if i keep the cursor on another window, no matter how i click on the icon, it won't turn blue

**Comment by EpsilonKu** [comment]
Hmm, weird. Let's wait @rcarriga

**Comment by st0nie** [comment]
![Screenshot_20230404_124950](https://user-images.githubusercontent.com/42872734/229689892-a34d1efa-0332-4e39-a5d6-5b092431e76b.png)
![Screenshot_20230404_125033](https://user-images.githubusercontent.com/42872734/229689900-e52e3c56-b4b3-4ded-9a3a-0e34049a140f.png)

Indeed, now that my cursor is on the left side of the dap-console, the icons are gray, and once I switch to the dap-repl window, they immediately turn blue, but by this time the program has stopped at the breakpoint

**Comment by st0nie** [comment]
minimal config
```lua
-- ignore default config and plugins
vim.opt.runtimepath:remove(vim.fn.stdpath("config"))
vim.opt.packpath:remove(vim.fn.stdpath("data") .. "/site")
vim.opt.termguicolors = true
vim.opt.laststatus = 0

-- append test directory
local test_dir = "/tmp/nvim-config"
vim.opt.runtimepath:append(vim.fn.expand(test_dir))
vim.opt.packpath:append(vim.fn.expand(test_dir))

-- install packer
local install_path = test_dir .. "/pack/packer/start/packer.nvim"
local install_plugins = false

if vim.fn.empty(vim.fn.glob(install_path)) > 0 then
        vim.cmd("!git clone https://github.com/wbthomason/packer.nvim " .. install_path)
        vim.cmd("packadd packer.nvim")
        install_plugins = true
end

local packer = require("packer")

packer.init({
        package_root = test_dir .. "/pack",
        compile_path = test_dir .. "/plugin/packer_compiled.lua",
})

packer.startup(function(use)
        -- Packer can manage itself
        use("wbthomason/packer.nvim")

        use({ "mfussenegger/nvim-dap" })

        use({
                "rcarriga/nvim-dap-ui",
                requires = "mfussenegger/nvim-dap",
                config = function()
                        local dap = require("dap")
                        dap.adapters.lldb = {
                                type = "executable",
                                command = "/usr/bin/lldb-vscode",
                                name = "lldb",
                        }
                        dap.configurations.cpp = {
                                {
                                        name = "Launch",
                                        type = "lldb",
                                        request = "launch",
                                        program = function()
                                                return vim.fn.input("Path to executable: ", vim.fn.getcwd() .. "/", "file")
                                        end,
                                        cwd = "${workspaceFolder}",
                                        stopOnEntry = false,
                                        args = {},
                                        runInTerminal = true,
                                },
                        }
                        dap.configurations.c = dap.configurations.cpp
                        dap.configurations.rust = dap.configurations.cpp
                        require("dapui").setup({})
                end,
        })

        if install_plugins then
                packer.sync()
        end
end)

vim.cmd([[
command! DapUIToggle lua require("dapui").toggle()
]])

```

**Comment by st0nie** [comment]
```
NVIM v0.8.3
Build type: Release
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/x86_64-pc-linux-gnu-gcc -march=native -O3 -pipe -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_TS_HAS_SET_MATCH_LIMIT -DNVIM_TS_HAS_SET_ALLOCATOR -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wdouble-promotion -Wmissing-noreturn -Wmissing-format-attribute -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/var/tmp/portage/app-editors/neovim-0.8.3/work/neovim-0.8.3_build/cmake.config -I/var/tmp/portage/app-editors/neovim-0.8.3/work/neovim-0.8.3/src -I/usr/include -I/var/tmp/portage/app-editors/neovim-0.8.3/work/neovim-0.8.3_build/src/nvim/auto -I/var/tmp/portage/app-editors/neovim-0.8.3/work/neovim-0.8.3_build/include
Compiled by portage@localhost

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "/etc/vim/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

**Comment by AlanJui** [comment]
I also encountered a similar problem. When I exited Neovim and returned to the command line, I saw the following error messages on the screen:

```
❯ nvim

Rendering failed: ...ack/packer/start/nvim-dap-ui/lua/dapui/render/canvas.lua:154: Vim:E315: ml_get: 錯誤的 lnum: 2
stack traceback:
...im/site/pack/packer/start/nvim-dap-ui/lua/dapui/util.lua:17: in function <...im/site/pack/packer/start/nvim-dap-ui/lua/dapui/util.lua:16>
[C]: in function 'nvim_buf_set_lines'
...ack/packer/start/nvim-dap-ui/lua/dapui/render/canvas.lua:154: in function 'render_buffer'
.../packer/start/nvim-dap-ui/lua/dapui/elements/watches.lua:70: in function 'render'
.../packer/start/nvim-dap-ui/lua/dapui/elements/watches.lua:24: in function <.../packer/start/nvim-dap-ui/lua/dapui/elements/watches.lua:23>
[C]: in function 'xpcall'
...im/site/pack/packer/start/nvim-dap-ui/lua/dapui/util.lua:16: in function <...im/site/pack/packer/start/nvim-dap-ui/lua/dapui/util.lua:12>%
```

**Comment by AlanJui** [comment]
```
:version
NVIM v0.8.3
Build type: Release
LuaJIT 2.1.0-beta3
編譯者:brew@Ventura

Features: +acl +iconv +tui
See ":help feature-compile"

        系統 vimrc 設定檔: "$VIM/sysinit.vim"
              $VIM 預設值: "/usr/local/Cellar/neovim/0.8.3/share/nvim"

Run :checkhealth for more info

```

**Comment by st0nie** [comment]
solved the issue by upgrade to NVIM v0.9.0
